adapters/huaweiads/huaweiads.go

```diff
@@ -935,12 +940,12 @@ func getReqRegsInfo(request *huaweiAdsRequest, openRTBRequest *openrtb2.BidReque
 // getReqGeoInfo: get geo information for HuaweiAds request, include Lon, Lat, Accuracy, Lastfix
 func getReqGeoInfo(request *huaweiAdsRequest, openRTBRequest *openrtb2.BidRequest) {
 	if openRTBRequest.Device != nil && openRTBRequest.Device.Geo != nil {
-		var geo geo
-		geo.Lon = float32(openRTBRequest.Device.Geo.Lon)
-		geo.Lat = float32(openRTBRequest.Device.Geo.Lat)
-		geo.Accuracy = int32(openRTBRequest.Device.Geo.Accuracy)
-		geo.Lastfix = int32(openRTBRequest.Device.Geo.LastFix)
-		request.Geo = geo
+		request.Geo = geo{
+			Lon:      float32(ptrutil.ValueOrDefault(openRTBRequest.Device.Geo.Lon)),
+			Lat:      float32(ptrutil.ValueOrDefault(openRTBRequest.Device.Geo.Lat)),
```

Comment:
The range for Lat is -90 to 90 and the range for Lon is -180 to 180. Neither approach the maximum size for float32, so the conversion from float64 to float32 may just reduce precision. If a value is so large that it may wrap around, it's invalid data.